CameraJoint.gd

```diff
@@ -0,0 +1,31 @@
+extends SpringArm
+
+
+var mouse_sensitivity = 0.5
+var scroll_speed = 40
+
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	set_as_toplevel(true)
+	spring_length = 7
+
+#func _process(delta):
+#	pass
+
+func _physics_process(delta):
+	# zoom the camera in or out based on the scroll wheel
+	if Input.is_action_just_released("scroll_up"):
+		spring_length -= scroll_speed * delta
+	if Input.is_action_just_released("scroll_down"):
+		spring_length += scroll_speed * delta
+	spring_length = clamp(spring_length, 0, 12)
+
+func _unhandled_input(event):
+	# move the camera if the RMB is held
+	if event is InputEventMouseMotion and Input.is_action_pressed("right_mouse_button"):
+		rotation_degrees.y -= event.relative.x * mouse_sensitivity
+		rotation_degrees.x = clamp(rotation_degrees.x, -90.0, 90.0)
+		
+		rotation_degrees.x -= event.relative.y * mouse_sensitivity
+		rotation_degrees.y = wrapf(rotation_degrees.y, 0.0, 360.0)
```

Comment:
This looks like it might be a standard pattern/behavior, but it's not intuitive.  Can you add comments and/or make it a function with a clear name?

Reply:
It took me awhile to revisit [where I got this code from](https://youtu.be/UpF7wm0186Q?t=290) to figure out what it's doing, but I finally figured it out. The thing that makes this more confusing is that lines 28 and 31 should be swapped. 

Anyway, the idea here is that changing `rotation_degrees.y` means rotating around the y-axis, which means that this will rotate the camera left or right. Then, we need to make the new y rotation be within the bounds of 0 and 360 degrees (line 31). It's the same concept for `rotation_degrees.x` except that we cap the rotation to +90 and -90 so the player can't rotate the camera so far up or down.

Reply:
Swapping those lines will probably help the readability too